api/src/services/survey-service.ts

```diff
@@ -491,31 +489,52 @@ export class SurveyService extends DBService {
   }
 
   /**
-   * To update a survey permit, we need to unassociate (not delete) the old permit from the survey and then associate
-   * the new permit to the survey. Associating a new permit to the survey is done by either inserting a brand new
-   * permit record into the permit table and setting the survey id column OR updating an existing permit record by
-   * setting the survey id column.
+   * Compares incoming survey permit data against the existing survey permits, if any, and determines which need to be
+   * deleted, added, or updated.
    *
-   * @param {number} projectId
    * @param {number} surveyId
    * @param {PutSurveyObject} surveyData
-   * @return {*}
    * @memberof SurveyService
    */
-  async updateSurveyPermitData(projectId: number, surveyId: number, surveyData: PutSurveyObject) {
-    await this.unassociatePermitFromSurvey(surveyId);
+  async updateSurveyPermitData(surveyId: number, surveyData: PutSurveyObject) {
+    const permitService = new PermitService(this.connection);
 
-    if (!surveyData.permit.permit_number) {
-      return;
+    // Get existing permits
+    const existingPermits = await permitService.getPermitBySurveyId(surveyId);
+
+    // Determine if any existing permits need to be deleted (are not included in the incoming permits array)
```

Comment:
comment is a little confusing maybe change to
'Incoming permits array is our source of truth, find and remove any hanging/ leftover permits'

Reply:
Took a stab at updating all of these comments to try and be more descriptive, not sure if its better or just different :)

Reply:
I think it's better!